aten/src/ATen/cuda/Atomic.cuh

```diff
@@ -484,7 +460,11 @@ __host__ __device__ T safe_min(T a, T b) {
 }
 
 ATOMIC_INTEGER_IMPL(Min)
-GPU_ATOMIC_INTEGER(Min, safe_min(a, b))
```

Comment:
looking at this now, I realize that we actually don't need to use safe_min/safe_max which does nan checks for integers because integer types do not have NANs iiuc (which is my bad from the previous PR).

I believe the windows compiler will just complain if we just pass std::min(a, b) so we need std::min<DTYPE, DTYPE>(a, b). Out of curiosity, would you expect this to work if we passed std::min<DTYPE, DTYPE>(a, b) to the macro? (I can fix this in a followup)

Reply:
I think the template types can be deduced from the arguments. I tried playing with it a bit in godbolt: https://godbolt.org/z/P8PxProj1